--- a/taskvine/src/manager/vine_manager.c
+++ b/taskvine/src/manager/vine_manager.c
@@ -1402,7 +1402,7 @@
 			w->addrport,
 			(t->time_when_done - t->time_when_commit_start) / 1000000.0,
 			(long long)w->total_tasks_complete,
-			w->total_task_time / w->total_tasks_complete / 1000000.0);
+			w->total_tasks_complete > 0 ? (double)w->total_task_time / w->total_tasks_complete / 1e6 : 0.0);
 
 	return;
 }
